include/kalmar_runtime.h

```diff
@@ -245,7 +264,7 @@ class KalmarQueue
   /// enqueue marker with prior dependency
   virtual std::shared_ptr<KalmarAsyncOp> EnqueueMarkerWithDependency(int count, std::shared_ptr <KalmarAsyncOp> *depOps, memory_scope scope) { return nullptr; }
 
-  virtual std::shared_ptr<KalmarAsyncOp> detectStreamDeps(hcCommandKind commandKind, KalmarAsyncOp *newCopyOp) { return nullptr; };
+  virtual std::shared_ptr<KalmarAsyncOp> detectStreamDeps(hcCommandKind commandKind, KalmarAsyncOp *newCopyOp, memory_scope *requiredRelease) { return nullptr; };
```

Comment:
@bensander It seems the 3rd parameter, `requiredRelease`, is getting much more importance in this PR. Would you mind add more comments here to explain how the function works?

And would it be possible to change `requiredRelease` have be optional, like `memory_scope *requiredRelease = nullptr`? Because it seems to me not all call sites of `detectStreamDeps` really cares about the `memory_scope` changed by it.